packages/core/tests/scope-directive.spec.ts

```diff
@@ -0,0 +1,367 @@
+import { expect, use } from 'chai';
+import { AtRule, Declaration, Rule } from 'postcss';
+import { processorWarnings } from '../src';
+import { flatMatch } from './matchers/flat-match';
+import { expectWarnings, expectWarningsFromTransform } from './utils/diagnostics';
+import { generateStylableResult, generateStylableRoot, processSource } from './utils/generate-test-util';
+
+use(flatMatch);
+
+describe('@st-scope', () => {
+    describe('processing scopes', () => {
+        it('should parse "@st-scope" directives', () => {
+            const meta = processSource(
+                `
+                .root {}
+                @st-scope root {
+                    .part {}
+                }
+            `,
+                { from: 'path/to/style.css' }
+            );
+
+            expect(meta.scopes).to.flatMatch([{
+                type: 'atrule',
+                name: 'st-scope',
+                params: 'root'
+            }]);
+        });
+
+        it('should mark scope ref name on impacted rules', () => {
+            const meta = processSource(
+                `
+                .root {}
+                @st-scope root {
+                    .part {}
+                    .otherPart {}
+                }
+            `,
+                { from: 'path/to/style.css' }
+            );
+
+            const rules = meta.ast.nodes!;
+            expect((rules[1] as Rule).selector).to.equal('.root .part');
```

Comment:
why does it transform the AST at the process phase? shouldn't the change only be in the transtorm?

Reply:
By removing the syntax sugar during processing we're ending up with selectors that are still Stylable symbols (untransformed), this makes the processing result already partly usable.

Similar to how we behave with imports/vars/custom selectors, we remove our own custom syntax from the ast where possible, leaving the final transformation up to the transformer.